src/mo_types/type.ml

```diff
@@ -1479,6 +1479,42 @@ and singleton_field co tf = singleton_typ co tf.typ
 
 and singleton t : bool = singleton_typ (ref S.empty) t
 
+(** A type is isolated if it has no proper supertypes nor proper subtypes (ignoring top [Any] and bottom [Non]). *)
+let rec isolated_typ co t =
+  S.mem t !co || begin
+  co := S.add t !co;
+  match normalize t with
+  | Prim
+    ( Bool
+    | Nat8
+    | Nat16
+    | Nat32
+    | Nat64
+    | Int8
+    | Int16
+    | Int32
+    | Int64
+    | Float
+    | Char
+    | Text
+    | Blob
+    | Error
+    | Principal
+    | Region
+    (* All except Nat, Int, Null as they have proper super/subtypes: Nat <: Int, ?T <: Null *)
+    )
+  | Mut _ -> true
+  | Array t
+  | Async (_, _, t)
+  | Weak t -> isolated_typ co t
+  | Tup ts -> List.for_all (isolated_typ co) ts
+  | Func (_, _, _, ts1, ts2) ->
```

Comment:
I think you still want a case for  a type parameter (Con(t, ts) with kind c = Abs). This can be isolated iff the bound is Any (which is common) (I think).

I guess that means that

func box<T>( v : T) {
   var r = Box(v);
}
would work.
But

func box<T: Nat>( v : T) {
   var r = Box(v);
}
would not.